### PR TITLE
fix(telemetry): Use transactions for tmp telemetry DB

### PR DESCRIPTION
**Pull request by mhofman**
refs: #6056

## Description

Solves a performance issue where automatic transactions compounded to cause noticeable stalls.
Similar to https://github.com/Agoric/agoric-sdk/pull/6058 but for the telemetry SQLite DB.

### CODE REVIEW

**Comment by mhofman** [comment]
Probably not the right level of abstraction, but I'm not really willing to refactor things more than this.

**Review by warner — APPROVED** [review]
Sounds good to me.. thanks for remembering that we had another SQLite DB around.

I'm a bit weirded out that our telemetry is doing this much correlation work in-line. I'd like to revisit that and see if it's really pulling its weight.

To estimate what sort of load this used to represent.. it looks like we'd get one DB commit per slog `syscall.send` event (as long as a `result` vpid was provided), right?

**Comment by mhofman** [comment]
> Sounds good to me.. thanks for remembering that we had another SQLite DB around.

Credit to @arirubinstein for noticing

> I'm a bit weirded out that our telemetry is doing this much correlation work in-line. I'd like to revisit that and see if it's really pulling its weight.

Yeah I was a little surprised as well, and we probably should revisit how this is implemented, but this correlation can be beneficial when looking at traces on Honeycomb.

> 
> To estimate what sort of load this used to represent.. it looks like we'd get one DB commit per slog `syscall.send` event (as long as a `result` vpid was provided), right?

Correct. Maybe not as bad as every crank, but realistically there are quite a few send in each block.

**Comment by warner** [comment]
Looks like this derailed somehow.. I'll force-push a rebased version to resolve at least one of the problems.

File: packages/telemetry/src/kv-string-store.js [hunk 1]
```diff
@@ -22,6 +22,21 @@ export const makeTempKVDatabase = io => {
...
+  return {
+    begin: () => {
+      assert(!db.inTransaction);
+      db.prepare('BEGIN IMMEDIATE TRANSACTION').run();
+      assert(db.inTransaction);
+    },
+    end: () => {
+      assert(db.inTransaction);
+      db.prepare('COMMIT').run();
+      assert(db.inTransaction);
```
Comment (michaelfig) [inline packages/telemetry/src/kv-string-store.js]: Is this wrong?

```suggestion
      assert(!db.inTransaction);
```

**Review by michaelfig — COMMENTED** [review]
It looks like the transaction flag is being asserted when it isn't set.

File: packages/telemetry/src/kv-string-store.js (hunk 1 above)
Comment (warner) [reply packages/telemetry/src/kv-string-store.js]: you're absolutely right.. I pushed a patch to fix it

Comment (mhofman) [reply packages/telemetry/src/kv-string-store.js]: Yikes 😞

**Pull request merged**